gnosis-imap/gnosis-imap-spec.ini

```diff
@@ -148,6 +148,138 @@ proxy_storage:
+;callValue: 0
+;this: PROXY_ID
+;proxy_storage:
+;    M1:IMap => M2:IMap
+;msg_sender: MSG_SENDER
+;+requires:
+;   andBool #rangeAddress(MSG_SENDER)
+;   andBool #rangeAddress(PROXY_ID)
+;   andBool #rangeAddress(OWNER)
+;   andBool #rangeUInt(256, NEW_THRESHOLD)
+;   andBool select(M1, #hashedLocation({COMPILER}, {THRESHOLD},  .IntList)) ==Int THRESHOLD
+;   andBool select(M1, #hashedLocation({COMPILER}, {OWNERCOUNT}, .IntList)) ==Int OWNERCOUNT 
+;   andBool select(M1, #hashedLocation({COMPILER}, {OWNERS},     1))        ==Int OLD_OWNER
+;   andBool #rangeUInt(256, THRESHOLD) 
+;   andBool #rangeUInt(256, OWNERCOUNT)
+;   andBool #rangeAddress(OLD_OWNER)
+;
+;[addOwnerWithThreshold-success]
+;statusCode: _ => EVMC_SUCCESS
+;log: _:List ( .List => ListItem(#abiEventLog(MSG_SENDER, "AddedOwner", #address(OWNER))) )
+;+requires:
+;   andBool select(M1, #hashedLocation({COMPILER}, {OWNERS}, OWNER)) ==Int 0
+;   andBool OWNER    =/=Int 0
+;   andBool OWNER    =/=Int 1
+;   andBool PROXY_ID ==Int MSG_SENDER
+;   andBool #rangeUInt(256, OWNERCOUNT +Int 1) 
+;
+;
+;[addOwnerWithThreshold-success-1]
+;+requires:
+;   andBool NEW_THRESHOLD ==Int THRESHOLD
+;ensures:
+;   ; ensures M2 ==IMap store(store(store(store(M1, #hashedLocation({COMPILER}, {OWNERS}, 1), OWNER), #hashedLocation({COMPILER}, {OWNERS}, OWNER), OLD_OWNER), #hashedLocation({COMPILER}, {THRESHOLD}, .IntList), THRESHOLD), #hashedLocation({COMPILER}, {OWNERCOUNT}, .IntList), OWNERCOUNT +Int 1)  
+;
+;[addOwnerWithThreshold-success-2]
+;callStack: _ => #abiCallData("changeThreshold", (#uint256(NEW_THRESHOLD)))
+;+requires:
+;   andBool NEW_THRESHOLD =/=Int THRESHOLD
+;ensures:
+;  ; ensures M2 ==IMap store(store(store(store(M1, #hashedLocation({COMPILER}, {OWNERS}, 1), OWNER), #hashedLocation({COMPILER}, {OWNERS}, OWNER), OLD_OWNER), #hashedLocation({COMPILER}, {THRESHOLD}, .IntList), _), #hashedLocation({COMPILER}, {OWNERCOUNT}, .IntList), OWNERCOUNT +Int 1)  
+;
+;[addOwnerWithThreshold-failure-1]
+;statusCode: _ => EVMC_REVERT
+;log: _
+;+requires:
+;    andBool ( PROXY_ID =/=Int MSG_SENDER 
+;      orBool  OWNER    ==Int 0
+;      orBool  OWNER    ==Int 1 
+;      orBool  select(M1, #hashedLocation({COMPILER}, {OWNERS}, OWNER)) =/=Int 0 )
+;ensures:
+;
+;[addOwnerWithThreshold-failure-2]
+;statusCode: _ => EVMC_INVALID_INSTRUCTION
+;log: _
+;+requires:
+;    andBool (OWNERCOUNT +Int 1) >=Int (2 ^Int 256)
+;    andBool OWNER =/=Int 0
+;    andBool OWNER =/=Int 1
+;ensures:
+
+[addOwnerWithThreshold]
+k: (#execute => #halt) ~> _
+callStack: _
+wordStack: .WordStack => _
+localMem: .IMap => _
+pc: 0 => _
+gas: #gas(STARTGAS, 0, 0) => _
+memoryUsed: 0 => _
+log: _ => _
+refund: _ => _
+coinbase: _ => _
+output: _ => _
+this: #PROXY_ID
+msg_sender: #PROXY_ID
+callData: #abiCallData("addOwnerWithThreshold", (
+            #address(OWNER), 
+            #uint256(NEW_THRESHOLD) ))
+callValue: 0
+proxy_storage:
+    M1:IMap => M2:IMap
```

Comment:
@ana-pantilie I'm not positive for this approach. Why don't you try my approach that explicitly specifies the `store` terms in the LHS?  Please let me know if you have a specific reason to use this approach.